src/Compilers/Core/CodeAnalysisTest/Text/SourceTextTests.cs

```diff
@@ -212,6 +212,81 @@ public void ContentEquals()
+
+            var randomText = builder.ToString();
+
+            var allSourceTexts = new List<SourceText>();
+
+            foreach (var encoding in encodings)
+            {
+                // Make normal StringText objects that just wrap the string.
+                allSourceTexts.Add(new StringText(randomText, encodingOpt: encoding));
+
+                // Make LargeText versions that break the strings up into chunks.  This will help ensure that regardless
+                // of which SourceText we have that equality works properly.
+                foreach (var algorithm in hashAlgorithms)
+                    allSourceTexts.Add(LargeText.Decode(new StringReader(randomText), randomText.Length, encoding, algorithm));
+            }
+
+            foreach (var sourceText1 in allSourceTexts)
+            {
+                foreach (var sourceText2 in allSourceTexts)
+                    Assert.True(sourceText1.ContentEquals(sourceText2));
+            }
+        }
+
+        [Fact]
+        public void ContentEqualUnaffectedByEncoding1()
+        {
+            var encodings = new[] { null, Encoding.ASCII, new UTF8Encoding(encoderShouldEmitUTF8Identifier: true), new UTF8Encoding(encoderShouldEmitUTF8Identifier: false) };
+
+            foreach (var encoding1 in encodings)
+            {
+                var sourceText1 = new StringText("；", encoding1); // chinese semicolon
+
+                foreach (var encoding2 in encodings)
+                {
+                    var sourceText2 = new StringText("?", encoding2); // what a non-ascii char will map to.
+                    Assert.False(sourceText1.ContentEquals(sourceText2));
+
+                    // Even though the contents are clearly different, they have the same checksum in ascii because of
+                    // the lossy mapping.  This shows how checksums should not be used to validate content equality.
+                    if (encoding1 == Encoding.ASCII && encoding2 == Encoding.ASCII)
+                        Assert.True(sourceText1.GetChecksum().SequenceEqual(sourceText2.GetChecksum()));
+                }
+            }
+        }
+
+        [Fact(Skip = "https://github.com/dotnet/roslyn/issues/70752")]
+        public void ContentEqualUnaffectedByEncoding2()
+        {
+            var sourceText1 = new StringText("；", Encoding.ASCII); // chinese semicolon
+            var sourceText2 = new StringText("?", Encoding.ASCII); // what a non-ascii char will map to.
+
+            var checksum1 = sourceText1.GetChecksum();
+            var checksum2 = sourceText2.GetChecksum();
+
+            Assert.True(checksum1.SequenceEqual(checksum2));
+
+            // This should return false, but is thrown off by the calls to GetChecksum above.  This is a bad bug that
+            // will be fixed once we resolve https://github.com/dotnet/roslyn/issues/70752
+            Assert.False(sourceText1.ContentEquals(sourceText2));
```

Comment:
This assert currently fails.  Which is BAD.  This test shows how though even though we have entirely different contents, the improper use of checksums makes us think the contents are actually the same.